doc/man/man3/xhostent.3.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) $Id$
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation; with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.  A
.\" copy of the license is included in the section entitled "GNU Free
.\" Documentation License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaning to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF
.\" SUCH CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified $Date$ by $Author$
.\"
.\" -------------------------------------------------------------------------
.\"
.\" $Log$
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore T
.R2
.so openss7.macros
.\"
.\"
.TH XHOSTENT 3 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Library Subroutines"
.\" =========================================================================
.SH NAME
.B endxhostent, getxhostbyaddr, getxhostbyname, getxhostent, setxhostent
\- X.25 Host related functions
.\"
.\"
.SH SYNOPSIS
.PP
.HP
.BI endxhostent
.HP
.BI getxhostbyaddr
.HP
.BI getxhostbyname
.HP
.BI getxhostent
.HP
.BI setxhostent
.XX
.[
sunlink8
.]
.[
sunlink9
.]
.[
$LIST$
.]


NEWS
OpenSS7 Master Package -- history of user visible changes.  2009-04-04
$Id: NEWS,v 0.9.2.72 2009-04-19 00:28:25 brian Exp $
Copyright (c) 2008-2009  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
`OpenSS7 Master Package' package.

 Major changes for release openss7-0.9.2.F.20090405
---------------------------------------------------

This is the seventh `OpenSS7 Project' release of the `OpenSS7 Master
Package' package.  For a sub-package listing, see `README'
(http://www.openss7.org/codefiles/openss7/README) in the release.  Most
of the sub-packages in the release are production grade for `Linux
Fast-STREAMS'.(1)  All existing validation test suites run clean on
supported distributions, production kernels and architectures.

The original intention was that the `OpenSS7 Master Package' package
not be released as often as subpackages and, indeed, newer subpackages
can be strapped into the master package and built automatically.
However, in reality, the `OpenSS7 Master Package' package has been
released as often as the subpackages.  To rebuild the master package
with a new sub-package release, simply replace the directory to which
the sub-package belongs with the unpackged sub-package reelase and then
rebuild the master package.  This release provides support for recent
distributions, production kernels and tool chains.  The release also
provides support for significiant new features in the sub-packages.

Major features since the last public release are:

   - Addition of the `stratm' sub-package to the master build and
     release that now includes all of the ATM components that were
     previously part of the `strss7' sub-package.

   - Significant rework of the `strchan' sub-package into the `strss7',
     `strisdn', `strx25' and `striso' packages.  One single driver now
     meets the needs of all sub-packages.

   - Completion of the `strsock' package, moves from pre-alpha to beta.

   - The `strnsl' subpackage moves from alpha to production.

   - Significant completion of the `sigtran' package: all components
     moved back to production.


_This is a public, stable, production grade release of the package: it
deprecates previous releases.  Please upgrade to the current release
before reporting bugs._

As with other OpenSS7 releases, this release configures, compiles,
installs and builds RPMs and DEBs for a wide range of Linux 2.4 and 2.6
`rpm(8)' and `dpkg(1)' based distributions, and can be used on
production kernels without patching or recompiling the kernel.

This package is publicly released strictly under the `GNU Affero
General Public License Version 3'; however, commercial licensing is
available.  The release is available as an `autoconf(1)' tarball, SRPM,
DSC, and a set of binary RPMs and DEBs for popular distributions.  See
the package page (http://www.openss.org/openss7_pkg.html) for the
`autoconf(1)' tarballs, SRPMs, DSCs and binary RPMs and DEBs.  See the
repository page (http://www.openss7.org/repos.html) for information
concerning network installation and update sources.

See
`http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/ChangeLog'
and `http://www.openss7.org/codefiles/openss7-0.9.2.F.20090405/NEWS' in
the release for more information.  Also, see the `openss7.pdf' manual
in the release (also in html
(http://www.openss7.org/openss7_manual.html)).

For the news release, see `http://www.openss7.org/rel20081029_L.html'.

---------- Footnotes ----------

(1) Sub-packages are no longer supported on the deprecated `LiS' STREAMS
package.

 Major changes for release openss7-0.9.2.G
------------------------------------------

This is the sixth public release of the OpenSS7 Master Package.  See
`README' in the release for a sub-package listing.  Most of the
sub-packages in the release are production grade for `Linux
Fast-STREAMS'.  All existing validation test suites run clean on
supported distributions and architectures.

The `OpenSS7 Master Package' is not released as often as the
sub-packages.  As sub-packages are released more often, to rebuild the
master package with a new sub-package release, simply replace the
directory to which the sub-package belongs with the unpacked
sub-package release and then rebuild the master package.  This release
provides support for recent distributions and tool chains.

Major features since the last public release are as follows:

   - License upgrade to `AGPL Version 3'.

   - Modifications to build under `Fedora' `2.6.22.5-49' kernel.  These
     changes also support `2.6.22.9-91.fc7' kernel.  Modifications to
     build under `Fedora' `2.6.25-45.fc9' and `2.6.26.5-45.fc9' kernels.
     Documented `lib32gcc1' problem on Ubuntu.  Noted problem running
     under XEN kernels.  XEN kernels are not yet supported.  Added
     `MODULE_VERSION' to all modules and drivers.

   - Ability to strap out major documentation build and installation
     primarily for embedded targets.  Improvements to common build
     process for embedded and cross-compile targets.  Cross-compile
     fixes (strap out `AC_FUNC_REALLOC' macro when cross-compiling).
     Conversion of RPM spec files to common approach for major
     subpackages.

     Build system now builds `yum(8)' repositories for RPMs and
     `apt-get(8)' repositories for DEBs.  Installation documentation
     has been updated to include details of repository install
     sourcesref.

   - Higher performance and updated performance papers.

   - Updated tool chain to `m4-1.4.12', `autoconf-2.63' and
     `texinfo-4.13'.  Support for `flex 2.5.33' in maintainer mode.

   - Updated references database for manual pages and roff documents.

   - Added the following major sub-packages to the master build (and
     release):


src/lib/xhostent.c
/*****************************************************************************

 @(#) $RCSfile$ $Name$($Revision$) $Date$

 -----------------------------------------------------------------------------

 Copyright (c) 2008-2009  Monavacon Limited <http://www.monavacon.com/>
 Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
 Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>

 All Rights Reserved.

 This program is free software: you can redistribute it and/or modify it under
 the terms of the GNU Affero General Public License as published by the Free
 Software Foundation, version 3 of the license.

 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
 details.

 You should have received a copy of the GNU Affero General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>, or
 write to the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA
 02139, USA.

 -----------------------------------------------------------------------------

 U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
 behalf of the U.S. Government ("Government"), the following provisions apply
 to you.  If the Software is supplied by the Department of Defense ("DoD"), it
 is classified as "Commercial Computer Software" under paragraph 252.227-7014
 of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
 successor regulations) and the Government is acquiring only the license rights
 granted herein (the license rights customarily provided to non-Government
 users).  If the Software is supplied to any unit or agency of the Government
 other than DoD, it is classified as "Restricted Computer Software" and the
 Government's rights in the Software are defined in paragraph 52.227-19 of the
 Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
 the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
 (or any successor regulations).

 -----------------------------------------------------------------------------

 Commercial licensing and support of this software is available from OpenSS7
 Corporation at a fee.  See http://www.openss7.com/

 -----------------------------------------------------------------------------

 Last Modified $Date$ by $Author$

 -----------------------------------------------------------------------------

 $Log$
 *****************************************************************************/

#ident "@(#) $RCSfile$ $Name$($Revision$) $Date$"

static char const ident[] = "$RCSfile$ $Name$($Revision$) $Date$";

#include <netx25/x25_proto.h>
#include <netx25/xnetdb.h>
#include <string.h>

/** @brief compare two X.25 addresses
  * @param		x1 a pointer to the structure containing the first X.25
  *			address for checking
  * @param x2		a pointer to the structure containing the second X.25
  *			address for checking
  * @version SX25_1.0
  * @par Alias:
  * This function is an implementation of equalx25().
  *
  * Compares two X.25 addresses by checking to see whether the two xaddrf
  * structures holding them are the same.
  *
  * Returns true if the two structures are the same and false if they are not.
  */
int
__sx25_equalx25(struct xaddrf *x1, struct xaddrf *x2)
{
	if (x1 == x2)
		goto match;
	if (x1->link_id != x2->link_id)
		goto nomatch;
	if (x1->aflags != x2->aflags)
		goto nomatch;
	if (x1->nsap_len != x2->nsap_len)
		goto nomatch;
	if (x1->DTE_MAC.lsap_len != x2->DTE_MAC.lsap_len)
		goto nomatch;
	if (strncmp((char *) x1->DTE_MAC.lsap_add,
		    (char *) x2->DTE_MAC.lsap_add, x1->DTE_MAC.lsap_len))
		goto nomatch;
	if (strncmp((char *) x1->NSAP, (char *) x2->NSAP, x1->nsap_len))
		goto nomatch;
      match:
	/* everything matched */
	return (1);
      nomatch:
	return (0);
}

/** @fn int equalx25(struct xaddrf *x1, struct xaddrf *x2)
  * @param x1
  * @param x2
  * @version SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_equalx25().
  */
#pragma weak equalx25
__asm__(".symver __sx25_equalx25,equalx25@@SX25_1.0");

/** @brief convert X.25 address to xaddrf structure.
  * @param cp points to a character string containing the X.25 address for
  * conversion.
  * @param xad points to the xaddrf structure containing the X.25 dot format
  * address.
  * @param lookup determines the level of address checking carried out bfore
  * the address is converted.
  * @version SX25_1.0
  * @par Alias:
  * This function is an implementation of stox25().
  *
  * Converts an X.25 format address into an xaddrf structure.  Can also be used
  * as a validity check for X.25 addresses.
  *
  * When @lookup is zero (0), no address checking is carried out.  This allows
  * for faster conversion, but means the address may not be valid for the type
  * of network on which it is used.  A non-zero value of @lookup means that the
  * address format is checked with the configruation file for the link.
  *
  * Returns zero (0) upon successful completion.  Returns a negative value when
  * the X.25 address is invalid.
  */
int
__sx25_stox25(unsigned char *cp, struct xaddrf *xad, int lookup)
{
	return (-1);
}

/** @fn int stox25(unsigned char *cp, struct xaddrf *xad, int lookup)
  * @param cp
  * @param xad
  * @param lookup
  * @version SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_stox25().
  */
#pragma weak stox25
__asm__(".symver __sx25_stox25,stox25@@SX25_1.0");

/** @brief convert xaddrf structure to X.25 address
  * @param xad points to the xaddrf structure for conversion.
  * @param cp points to the string the X.25 address will be written to.
  * @param lookup determines the leve of address checking carried out before
  * the structure is converted.
  * @version SX25_1.0
  * @par Alias:
  * This function is an implementation of x25tos().
  *
  * Converts an xaddrf structure into an X.25 address.  Before doing so, it
  * checks the validity of the xaddrf structure.
  *
  * A @lookup value of zero (0) indicates no checking is carried out.  This
  * allows for faster conversion, but means the structure may not be valid for
  * the type of network to which it refers.  A non-zero value means that the
  * structure is checked using the configuration files.
  *
  * Zero (0) indicates successful completion.  A negative return value
  * indicates that the structure was invalid.
  */
int
__sx25_x25tos(struct xaddrf *xad, unsigned char *cp, int lookup)
{
	return (-1);
}

/** @fn int x25tos(struct xaddrf *xad, unsigned char *cp, int lookup)
  * @param xad
  * @param cp
  * @param lookup
  * @version SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_x25tos().
  */
#pragma weak x25tos
__asm__(".symver __sx25_x25tos,x25tos@@SX25_1.0");

/** @brief
  * @param stayopen
  * @verion SX25_1.0
  * @par Alias:
  * This function is an implementation of setxhostent();
  */
void
__sx25_setxhostent(int stayopen)
{
}

/** @fn void setxhostent(int stayopen)
  * @param stayopen
  * @verion SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_setxhostent().
  */
#pragma weak setxhostent
__asm__(".symver __sx25_setxhostent,setxhostent@@SX25_1.0");

/** @brief get X.25 host name by address
  * @param addr		A pointer to an xaddrf strcuture contiaining the
  *			address of the host whose entry is to be retrieved.
  * @param len		The length in bytes of addr.
  * @param type		The address type required.  This is always CCITT_X25.
  * @verion SX25_1.0
  * @par Alias:
  * This function is an implementation of getxhostbyaddr();
  *
  * This function searches the xhosts file for an entry with a matching X.25
  * host address.  By default, this file is located in the /etc/x25/ directory.
  * It returns a pointer to an xhostent structure containing information on the
  * entry.
  *
  * Returns a pointer to static storage.  You must copy the value to keep and
  * reuse it.  A return value of NULL indicates the address suspplied is either
  * invalid or unknown.
  */
struct xhostent *
__sx25_getxhostbyaddr(char *addr, int len, int type)
{
	return (NULL);
}

/** @fn struct xhostent *getxhostbyaddr(char *addr, int len, int type)
  * @param addr
  * @param len
  * @param type
  * @verion SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_getxhostbyaddr().
  */
#pragma weak getxhostbyaddr
__asm__(".symver __sx25_getxhostbyaddr,getxhostbyaddr@@SX25_1.0");

/** @brief get X.25 address by name
  * @param name A pointer to the address of a string containing the name of the
  * host whose entry is to be retrieved.
  * @verion SX25_1.0
  * @par Alias:
  * This function is an implementation of getxhostbyname();
  *
  * This function searches the xhosts file for an entry with a matching host
  * name.  By default, this file is located in the /etc/x25/ directory.  It
  * returns a pointer to an xhostent structure containing information on the
  * entry.
  *
  * Returns a pointer to an xhostent structure.  A return value of NULL
  * indicates the name supplied is either invalid or unknown.
  */
struct xhostent *
__sx25_getxhostbyname(char *name)
{
	return (NULL);
}

/** @fn struct xhostent *getxhostbyname(char *name)
  * @param name
  * @verion SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_getxhostbyname().
  */
#pragma weak getxhostbyname
__asm__(".symver __sx25_getxhostbyname,getxhostbyname@@SX25_1.0");

/** @brief
  * @verion SX25_1.0
  * @par Alias:
  * This function is an implementation of getxhostent();
  *
  * This function reads the next line of the xhosts file, typically at
  * /etc/x25/xhosts.  It opens the file if necessary.
  *
  * Returns a pointer to an xhostent structure.  A return value of NULL
  * indicates an error.
  */
struct xhostent *
__sx25_getxhostent(void)
{
	return (NULL);
}

/** @fn struct xhostent *getxhostent(void)
  * @verion SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_getxhostent().
  */
#pragma weak getxhostent
__asm__(".symver __sx25_getxhostent,getxhostent@@SX25_1.0");

/** @brief
  * @verion SX25_1.0
  * @par Alias:
  * This function is an implementation of endxhostent();
  *
  * This function closes the xhosts file.  By default, this file is located in
  * the /etc/x25/ directory.
  */
void
__sx25_endxhostent(void)
{
	return;
}

/** @fn void endxhostent(void)
  * @verion SX25_1.0
  * @par Alias:
  * This symbol is a weak alias of __sx25_endxhostent().
  */
#pragma weak endxhostent
__asm__(".symver __sx25_endxhostent,endxhostent@@SX25_1.0");

// vim: com=sr0\:/**,mb\:*,ex\:*/,sr0\:/*,mb\:*,ex\:*/,b\:TRANS
